lib/src/rules/parameter_assignments.dart

```diff
@@ -88,9 +136,33 @@ class _Visitor extends SimpleAstVisitor {
       // Getters don't have parameters.
       parameterList.parameters.forEach((e) {
         if (node.body.isPotentiallyMutatedInScope(e.element)) {
-          rule.reportLint(e);
+          _reportIfSimpleParameterOrWithDefaultValue(e, node);
         }
       });
     }
   }
+
+  void _reportIfSimpleParameterOrWithDefaultValue(
```

Comment:
You don't need to traverse the list even once if the parameter is a required parameter or has a default value, and I suspect that those two cases cover a large majority of the parameters in real code. Please at least make that optimization.